docs/advanced_topics.rst

```diff
@@ -72,6 +72,25 @@ Built-in iterators use the second API to provide convenient wrappers for immedia
 
 It is not recommended to mix any of the aforementioned APIs together, because they follow different logic for output buffer lifetime management and the details of the process are subject to change without notice. Mixing the APIs may result in an undefined behavior, like a deadlock or attempt to access an invalid buffer.
 
+Sharding
+--------
+
+Sharding allows DALI to partition the dataset into nonoverlapping pieces that every separate DALI pipeline instance can work on. This addresses the problem of having a global and shared state that allows the distribution of training samples among the ranks. After each epoch, the DALI pipeline by default advances to the next shard to increase the entropy of data seen by this particular pipeline. The user may customize this behavior through ``stick_to_shard`` reader parameter.
+
+This however leads to problems when the dataset size is not divisible by the number of GPUs, and when the shard size is not divisible by the batch size. To address this problem and adjust behavior depending on the user needs DALI provides several options.
+
+The first is ``pad_last_batch`` which asks the reader to duplicate the last sample in the last batch so DALI won't read ahead data from the next shard when the batch doesn't divide its size. Also it makes sure that all pipelines return the same number of batches - when one is dividable by the batch size but others are bigger by one sample.
+
+As DALI is used in the Deep Learning Frameworks through dedicated iterators, they need to be aware of this padding and other reader properties as well. Let us look into the following Iterator options:
+
+- ``size`` - the size of the shard for given iterator (or sum of all shard sizes for all wrapped pipelines if there is more than one)
+- ``fill_last_batch`` - whether the last batch should be full no matter if shard size is divisible by the batch size
+- ``last_batch_padded`` - whether the data that is reminder between multiply of batch size and shard size consists of data from the next shard or duplicated dummy data
+- ``reader_name`` - if the user wants to avoid providing the above parameters and let the iterator query online the selected reader in the pipeline. This the recommended way as it is more flexible and accurate (takes into account that shards size for the given pipeline can differ epoch to epoch when shards are rotated)
```

Comment:
Maybe put it at the top and describe that for iterator user can either set the `reader_name`, or specify the stuff manually (and list the `size`, `fill_last_batch` ...).

Reply:
Done